plugin/dnstap/io.go

```diff
@@ -62,16 +62,14 @@ func (d *dio) dial() error {
 			Timeout: d.tcpTimeout,
 		}
 		conn, err = tls.DialWithDialer(dialer, "tcp", d.endpoint, config)
-		if err != nil {
-			return err
-		}
 	} else {
 		conn, err = net.DialTimeout(d.proto, d.endpoint, d.tcpTimeout)
-		if err != nil {
-			return err
-		}
```

Comment:
I prefer the error checks here, since it makes it immediately clear that the errors are handled.